__tests__/allTests.test.js

```diff
@@ -0,0 +1,233 @@
+        .then(({body}) =>{
+            //console.log(body)
+            expect(body.topics).toHaveLength(3)
+            body.topics.forEach((topic) => {
+               // console.log(topic)
+
+                expect(topic).toMatchObject({
+                    slug: expect.any(String),
+                    description: expect.any(String)
+                })
+            })
+
+        });
+    })
+})
+
+
+// I think we should have had api call all other
+describe('/api/', () => {
+    test('GET:200 sends an object with API endpoints to the client', async () => {
+      const response = await request(app).get('/api/').expect(200);
+      const body = response.body;
+  
+     // console.log(body);
+
+      expect(body).toBeInstanceOf(Object);
+  
+      expect(body).toHaveProperty('GET /api');
+      expect(body).toHaveProperty('GET /api/articles');
+      expect(body).toHaveProperty('GET /api/articles/:article_id');
+      expect(body).toHaveProperty('GET /api/topics');
+  
+      expect(body['GET /api']).toMatchObject({
+        description: expect.any(String),
+      });
+  
+      expect(body['GET /api/articles']).toMatchObject({
+        description: expect.any(String),
+        exampleResponse: expect.any(Object),
+        queries: expect.any(Array),
+      });
+
+      //console.log(body['GET /api/articles/:article_id'])
+  
+      expect(body['GET /api/articles/:article_id']).toMatchObject({
+        article_id: expect.any(String),
+        article_img_url: expect.any(String),
+        author: expect.any(String),
+        created_at: expect.any(String),
+        title: expect.any(String),
+        topic: expect.any(String),
+        votes: expect.any(Number),
+      });
+  
+        //console.log(body['GET /api/topics'])
+
+      expect(body['GET /api/topics']).toMatchObject({
+        description: expect.any(String),
+        exampleResponse: expect.any(Object),
+        queries: expect.any(Array),
+      });
+  
+      expect(body['GET /api/topics'].exampleResponse.topics).toBeInstanceOf(Array);
+      body['GET /api/topics'].exampleResponse.topics.forEach((topic) => {
+        expect(topic).toMatchObject({
+          slug: expect.any(String),
+          description: expect.any(String),
+        });
+      });
+    });
+  }
+);
+
+
+// fix this later - doesnt work
+describe('/api/articles/:articles_id', () => {
+    test('GET:200 sends an object of topics to the client when making a valid request', () => {
+        return request(app)
+        .get('/api/articles/')
+        .expect(200)
+        .then(({body}) =>{
+            let obj = body[0]
+          //  console.log(obj)
+
+            expect(obj).toMatchObject({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+              comment_count: expect.any(Number)
```

Comment:
To ensure the correct comment count is returned is accurate, you should ideally expect an exact number here